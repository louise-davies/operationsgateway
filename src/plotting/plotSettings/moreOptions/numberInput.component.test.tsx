import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import NumberInput from './numberInput.component';

describe('NumberInput', () => {
  it('should render without errors', () => {
    const view = render(<NumberInput />);
    expect(view).toMatchSnapshot();
  });

  it('should increment the value when the increment button is clicked', () => {
    render(<NumberInput max={2} value={1} />);
    const incrementButton = screen.getByRole('button', { name: 'increment' });
    const input = screen.getByRole('textbox');
    fireEvent.click(incrementButton);
    expect(input).toHaveValue('2');

    // should not increment beyond max
    fireEvent.click(incrementButton);
    expect(input).toHaveValue('2');
    expect(incrementButton).toHaveAttribute('aria-disabled', 'true');
  });

  it('should decrement the value when the decrement button is clicked', () => {
    render(<NumberInput min={0} value={1} />);
    const decrementButton = screen.getByRole('button', { name: 'decrement' });
    const input = screen.getByRole('textbox');
    fireEvent.click(decrementButton);

    expect(input).toHaveValue('0');

    // should not decrement beyond min
    fireEvent.click(decrementButton);
    expect(input).toHaveValue('0');
    expect(decrementButton).toHaveAttribute('aria-disabled', 'true');
  });
});
