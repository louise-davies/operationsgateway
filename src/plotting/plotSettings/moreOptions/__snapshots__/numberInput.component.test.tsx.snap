// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberInput should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiNumberInput-root css-zmtyut"
      >
        <button
          aria-controls=":r0:"
          aria-disabled="false"
          aria-label="decrement"
          class="MuiNumberInput-decrementButton decrement css-1d3khx7"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m7 10 5 5 5-5z"
            />
          </svg>
        </button>
        <button
          aria-controls=":r0:"
          aria-disabled="false"
          aria-label="increment"
          class="MuiNumberInput-incrementButton increment css-1d3khx7"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="ArrowDropUpIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m7 14 5-5 5 5z"
            />
          </svg>
        </button>
        <input
          aria-disabled="false"
          aria-valuenow=""
          aria-valuetext=""
          autocomplete="off"
          autocorrect="off"
          class="MuiNumberInput-input css-3aul9a"
          id=":r0:"
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiNumberInput-root css-zmtyut"
    >
      <button
        aria-controls=":r0:"
        aria-disabled="false"
        aria-label="decrement"
        class="MuiNumberInput-decrementButton decrement css-1d3khx7"
        tabindex="-1"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m7 10 5 5 5-5z"
          />
        </svg>
      </button>
      <button
        aria-controls=":r0:"
        aria-disabled="false"
        aria-label="increment"
        class="MuiNumberInput-incrementButton increment css-1d3khx7"
        tabindex="-1"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="ArrowDropUpIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m7 14 5-5 5 5z"
          />
        </svg>
      </button>
      <input
        aria-disabled="false"
        aria-valuenow=""
        aria-valuetext=""
        autocomplete="off"
        autocorrect="off"
        class="MuiNumberInput-input css-3aul9a"
        id=":r0:"
        spellcheck="false"
        type="text"
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
